source/_posts/2018/2018-02-19-rector-part-2-why.md

```diff
@@ -0,0 +1,166 @@
+  
+That's why most of lectures (@todo link) I make is about giving teams *the knowledge* that they can use with their very limited resources to make the best out of it.
+ 
+That naturally leads to huge legacy code, team performance drop from 100 % to 20 %, which leads to hiring 5x more people etc. to keep productivity the same, more money, and pressure to faster development, which lead to huge legacy code...
+  
+### Are Deprecations Easy to Find?
+ 
+Let's say you have time to explore the Internet, follow [Symfony News on Twitter](https://twitter.com/symfony_en), read [every news post on Symfony Blog](http://symfony.com/blog/category/living-on-the-edge) or know where on the Nette forum are located [Release Notes](https://forum.nette.org/en/f78-release-announcements-news).
+
+Sometimes if you're lucky there is `UPGRADE-x.md` in project's Github repository, like [`UPGRADE-4.0.md`](https://github.com/symfony/symfony/blob/master/UPGRADE-4.0.md) in Symfony 4 repository. But what if you need upgrade to version 3.x? Could you find it? Well no, but yes in [3.x branch](https://github.com/symfony/symfony/tree/3.4).
+
+Sometimes these changes are in files called `CHANGELOG-x.md`.
+
+But more often they're newer to be seen and you have to go git blame in Github specific line and hope for an answer. And just pray that there is PR with more detailed changes with tests as well and not direct set of commits to the `master` branch without context.
+
+### When We Find Them, are They Valid?
+
+Sometimes there can be bare useless description, like [in `UPGRADE-4.0`](https://github.com/symfony/symfony/blob/master/UPGRADE-4.0.md#process):
+
+*The `Symfony\Component\Process\ProcessBuilder` class has been removed, use the `Symfony\Component\Process\Process` class directly instead.*
+
+So it easy as:
+
+```diff
+-use Symfony\Component\Process\ProcessBuilder;
++use Symfony\Component\Process\Process;
+
+-$builder = new ProcessBuilder();
++$builder = new Process();
+ $builder->setArguments(['build', '-force', '-var "blah=blah"', 'path/to/json.json'])
+    ->getProcess();
+```
+
+Unfortunately no and our investigative programming begins. Git blame..., Google? Symfony Docs?
+
+Sometimes they are embodied in the code - the best place to add them. When running this method, you'll be informed: 
+ 
+```php
+public function add()
+{
+    trigger_error('Method add() is deprecated, use addHtml() instead.', E_USER_DEPRECATED);
+}
+```
+
+### Found them and Valid! But are They Standardized?
+
+Sometimes the maintainer goes as far to deprecate code slowly (!= remove):
+
+```php
+trigger_error('Method add() is deprecated, use addHtml() instead.', E_USER_DEPRECATED);
+```
+
+Which is good enough for start. Symfony even has [PHPUnit Bridge](https://symfony.com/doc/current/components/phpunit_bridge.html) that tries to detect those deprecations. Would you know what exactly do you need to change?
+
+<img src="https://symfony.com/doc/current/_images/report.png">
```

Comment:
This `_image` is italicising all the text

Reply:
It's just the `_` sign, no problem